src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-configuration-page/rgw-configuration-page.component.ts

```diff
@@ -0,0 +1,147 @@
+  readonly vaultAddress = /^((https?:\/\/)|(www.))(?:([a-zA-Z]+)|(\d+\.\d+.\d+.\d+)):\d{4}$/;
+
+  kmsProviders: string[];
+
+  columns: Array<CdTableColumn> = [];
+
+  configForm: CdFormGroup;
+  permissions: Permissions;
+  encryptionConfigValues: any = [];
+  selection: CdTableSelection = new CdTableSelection();
+
+  @Output()
+  submitAction = new EventEmitter();
+  authMethods: string[];
+  secretEngines: string[];
+  tableActions: CdTableAction[];
+  bsModalRef: NgbModalRef;
+  filteredEncryptionConfigValues: {};
+  excludeProps: any[] = [];
+  disableCreate = false;
+  allEncryptionValues: any;
+
+  constructor(
+    public activeModal: NgbActiveModal,
+    public actionLabels: ActionLabelsI18n,
+    private rgwBucketService: RgwBucketService,
+    public authStorageService: AuthStorageService,
+    private rgwEncryptionModal: RgwBucketEncryptionModel,
+    private modalService: ModalService
+  ) {
+    super();
+    this.permissions = this.authStorageService.getPermissions();
+  }
+
+  ngOnInit() {
+    this.columns = [
+      {
+        name: $localize`Encryption Type`,
+        prop: 'encryption_type',
+        flexGrow: 1
+      },
+      {
+        name: $localize`Key Management Service Provider`,
+        prop: 'backend',
+        flexGrow: 1
+      },
+      {
+        name: $localize`Address`,
+        prop: 'addr',
+        flexGrow: 1
+      }
+    ];
+    const addAction: CdTableAction = {
+      permission: 'create',
+      icon: Icons.add,
+      name: this.actionLabels.CREATE,
+      click: () => this.openRgwConfigModal(false),
+      disable: () => this.disableCreateAction()
+    };
+    const editAction: CdTableAction = {
+      permission: 'update',
+      icon: Icons.edit,
+      name: this.actionLabels.EDIT,
+      click: () => this.openRgwConfigModal(true)
+    };
+    this.tableActions = [addAction, editAction];
+    this.rgwBucketService.getEncryptionConfig().subscribe((data: any) => {
+      this.allEncryptionValues = data;
+      const allowedBackends = this.rgwEncryptionModal.kmsProviders;
+      const kmsBackends = new Set();
+      data['SSE-KMS'].forEach((item: { backend: unknown }) => kmsBackends.add(item.backend));
+      const s3Backends = new Set();
+      data['SSE-S3'].forEach((item: { backend: unknown }) => s3Backends.add(item.backend));
+      const allKmsBackendsPresent = allowedBackends.every((backend) => kmsBackends.has(backend));
+      const allS3BackendsPresent = allowedBackends.every((backend) => s3Backends.has(backend));
+      if (allKmsBackendsPresent && allS3BackendsPresent) {
+        this.disableCreate = true;
+      } else {
+        this.disableCreate = false;
+      }
+      const flatConfigValues = Object.values(data).flat();
+      this.encryptionConfigValues = flatConfigValues;
+    });
+    this.excludeProps = this.columns.map((column) => column.prop);
+    this.excludeProps.push('unique_id');
+  }
+
+  disableCreateAction(): boolean {
+    return this.disableCreate;
+  }
```

Comment:
this function doesn't disable the creation action. it simply returns the value. I don't think we even need this function. You can just use the `this.disableCreate` variable wherever you want.